rts/Game/Camera/SpringController.cpp

```diff
@@ -54,13 +63,41 @@ void CSpringController::ConfigUpdate()
 	fastScaleMousewheel = configHandler->GetFloat("CamSpringFastScaleMousewheelMove");
 	doRotate = configHandler->GetBool("CamSpringEdgeRotate");
 	lockCardinalDirections = configHandler->GetBool("CamSpringLockCardinalDirections");
+	trackMapHeight = configHandler->GetInt("CamSpringTrackMapHeightMode");
+
+	if (trackMapHeight == HeightTracking::Smooth && !modInfo.enableSmoothMesh) {
+		LOG_L(L_ERROR, "Smooth mesh disabled");
+		trackMapHeight = HeightTracking::Terrain;
+	}
 }
 
 void CSpringController::ConfigNotify(const std::string & key, const std::string & value)
 {
 	ConfigUpdate();
 }
 
+void CSpringController::SmoothCamHeight(const float3& prevPos) {
+	if (!pos.IsInBounds()) {
+		return;
+	}
+
+	float3 camPos = GetPos(); // new camera pos with height from previous frame
+	camPos.y = std::max(camPos.y, CGround::GetHeightReal(camPos.x, camPos.z, false) + 5.0f);
+
+	// raycast to ground to simulate camera movement and find new point of focus
+	const float distToGround = CGround::LineGroundCol(camPos, camPos + dir * 150000.0f, false);
```

Comment:
consider some explanation for the magic 150k constant

Reply:
I think it's ok as is, the rest of the code using rays tracing is full of magic constants.